### ISSUE TITLE
Getting unexpected_error when trying to query Prometheus

### ISSUE THREAD

**Issue by omerlh** [samber/prometheus-query-js#11]
I just spent some time trying to figure why I cannot use this library to query my Prometheus instance. 
Look like for newest Prometheus version (I'm running 2.25.2), the gzip encoding is required.
This is what worked for me:
```
const promDriver = new PrometheusDriver({
  endpoint: "http://localhost:9090",
  baseURL: "/api/v1",
  headers: {
    'Accept-Encoding': 'gzip'
  },
  warningHook: (x => console.log(x))
});
```
This issue is more for future people who might face the same issue - maybe worth adding something on the readme?

**Comment by samber** [comment]
Hi @omerlh 

Can you tell me more about this issue, please?
I didn't find any information about such a requirement.

Are you using This library from a browser or into NodeJS? 🤔

Do you confirm this is not a proxy configuration in front of your prometheus server?

- reference to a Prometheus Github issue?
- some error message?

Thanks for your help ;)

**Comment by omerlh** [comment]
I am using NodeJS. Not using proxy, but using `kubectl port-forward` to access prom - not sure if related...

**Comment by samber** [comment]
`kubectl port-forward` is a simple tcp proxy. I don't expect any error here.

I suppose your proxy connection was broken at that time, and the lib responded with an "unexecpted_error" message.

I let the issue opened, in case we have a similar bug report in the future.

**Comment by omerlh** [comment]
Nope, this was very reproducible: as long as gzip encoding is not set, it was broken. Once I added it, it was solved...

**Comment by samber** [comment]
fixed v3.0.1

**Comment by samber** [comment]
hi @omerlh,

Since setting the Accept-Encoding header is forbidden, I wonder what browser are you using?

Looks like browsers automatically set this header.

**Comment by omerlh** [comment]
I am not using a browser... The code I wrote was running in nodeJS

**Comment by samber** [comment]
This is really weird. I cannot reproduce your error.

Can you start a local Prometheus instance ? `docker run --rm -it -p 9090:9090 prom/prometheus:v2.25.0`

Then: `npx ts-node examples/nodejs/app.ts` (replace the hostname at the top of the file)

I upgraded Axios yesterday. Can you try again please?

**Comment by omerlh** [comment]
TBH it has been a while since I worked on it, so I will try to find time to look into it but not promising anything :)

**Comment by RossComputerGuy** [comment]
I get the same problem with my "proxy" I made for my API, all I did was add the `Authorization` header to my user's access token. Works fine in curl but fails with the error message described here.

**Comment by RossComputerGuy** [comment]
I might have a solution, it just needs to be verified. You just need to append this to your `PrometheusDriver` configuration:
```ts
{
  requestInterceptor: {
    onFulfilled: (cfg) => ({
      ...cfg,
      responseType: 'json',
      decompress: false
    })
  }
}
```
